api/src/main/java/org/apache/iceberg/expressions/AggregateEvaluator.java

```diff
@@ -83,6 +83,18 @@ public void update(DataFile file) {
     }
   }
 
+  public void update(DataFile file, StructLike struct) {
```

Comment:
Is this new API needed? Why not update the existing API to have the `else if` case if it's a partition column?